### DISCUSSION TITLE
CSS metrics

### DISCUSSION THREAD

**Discussion by nbuy** [xoopscube/legacy#227] [General]
Gigamaster's new admin design so cool!

But some time broken that like this.
![Screenshot 2023-01-23 at 12-03-34 Admin Web Application Platform 管理メニュー システム設定](https://user-images.githubusercontent.com/1916482/213959448-c4637930-f974-4af2-81a6-7cf6cf1077ca.png)

Why like this happen.
It was depend width character is not same in language.
`<input size="22">` width depend using fontset, not seeing width.

So what is good choice in width metrics?
There is previus element, `<input type="number">` that width setting by css `max-width: 8ch`.
It looks good here. But that include same problem. The `ch` is unit of "0"'s font width.

The panel design come from outer limit width.
Then the metrics use absolute unit size in this case.
(or calcurate by left of other items?)

**Comment by gigamaster** [comment]
Sorry to answer only now. I did not get any notification from Github of new messages ( ? )
I have been merging XCL into xoopscube/legacy, admin theme is on my task list : 
- review the admin prototype grid / flex
- card and table
- cleanup CSS ( remove unused code and comments )
- admin preferences save to localstorage
Because we have several core modules and many templates (+ components), 
the goal is to make the admin theme minimalist (CSS variables) 
and responsive.
Thank you for your report. Let me know please if you find a bug or if you have a suggestion.
^_^/

**Reply by gigamaster** [reply]
![xcl-admin-metrics](https://user-images.githubusercontent.com/1905497/226136011-1ae08f86-19c2-41c0-b3de-05c292fc90f3.JPG)

#xoopscube #xcl23 #CSS #Metrics #Admin #Dev #Mode #Mobile
CSS Metrics - Admin ( dev mode )
Testing CSS metrics before merging update of themes and templates. to GitHub xoopscube/legacy
All core modules installed
Blocks installed MainMenu, UserMenu, Message, Themes, Tags
Custom blocks installed X-Update Store, Ads Banner 40ko, Custom Image-Slide 47ko
Default Theme #PicoCSS
Dev Preload and default JavaScript plugins
No cache
No compression
NOTE : Inject JavaScript ( e.g. notifications, CKEditor full bundle of plugins) is a context only available for Admins !

**Reply by gigamaster** [reply]
![xcl-public-mode-metrics](https://user-images.githubusercontent.com/1905497/226136135-edf6aed3-634b-438d-8b36-4803d218c987.JPG)
#xoopscube #xcl23 #CSS #Metrics #Admin #Dev #Mode #Mobile
CSS Metrics - Public view
Testing CSS metrics before merging update of themes and templates. to GitHub xoopscube/legacy
All core modules installed
Blocks installed MainMenu, UserMenu, Login, Themes, Tags
Custom blocks installed X-Update Store, Ads Banner 40ko, Custom Image-Slide 47ko
Default Theme #PicoCSS
Dev Preload and default JavaScript plugins
No cache
No compression
NOTE : Inject JavaScript ( e.g. notifications, CKEditor full bundle of plugins) is a context only available for Admins !

**Reply by gigamaster** [reply]
![xcl-public-desktop-metrics](https://user-images.githubusercontent.com/1905497/226136165-b51c6e4e-0189-4b13-86c7-6764ca403a21.JPG)
#xoopscube #xcl23 #CSS #Metrics #Admin #Dev #Mode #Desktop
CSS Metrics - Admin Public View Desktop
Testing CSS metrics before merging update of themes and templates. to GitHub xoopscube/legacy
All core modules installed
Blocks installed MainMenu, UserMenu, Message, Themes, Tags
Custom blocks installed X-Update Store, Ads Banner 40ko, Custom Image-Slide 47ko
Default Theme #PicoCSS
Dev Preload and default JavaScript plugins
No cache
No compression
NOTE :
1. Here we have used unconventional defer load of resources!
2. Inject JavaScript ( e.g. notifications, CKEditor full bundle of plugins) is a context only available for Admins !

**Reply by gigamaster** [reply]
XDesktop and XFury (mobile app) are not fully compatible with XOOPSCube/Legacy
The Render engine was customized for both versions. 
Since Xdesktop business logic is based on the principle "local first" 
It might be possible to merge some features into XCL23.

**Comment by nbuy** [comment]
So cool!

You have great sence that not have for me.

**Reply by nunoluciano** [reply]
> So cool!
> 
> You have great sence that not have for me.

@nbuy 
I have added some color-scheme to admin theme ( admin panel right ). 
So you can test and customize to fit your needs.
Maybe you can check the changes made to headerscript > class/Legacy_HeaderScript ?
It's very basic hack using native defer and preload ( framework-agnostic ).
There is also a JavaScript anf jQuery alternatives that we can look later.
Have fun ^_^/

**Reply by nunoluciano** [reply]
> So cool!
> 
> You have great sence that not have for me.

@nbuy 
Both themes are almost complete (front-end and back-end) 
Tomorrow, I will request a review of 2.3. RC
23-03-2023 is a good date for a release candidate 😊
